craft_parts/executor/part_handler.py

```diff
@@ -457,6 +457,32 @@ def _run_build(
                     stdout=stdout,
                     stderr=stderr,
                 )
+
+                # Organize the installed files as requested. We do this in the build step for
```

Comment:
This (useful) comment got moved and now it kind of looks like it only applies to this branch (the one with overlay visibility), which isn't true. How about keeping a single `organize_files()` call and only defining the parameters in this `if` branch?

Reply:
I do no like where the comment is either but this call to `organize_files` is not only in the `if` branch but more importantly under the `overlays.LayerMount` context because it needs the overlay to be mounted.  I will create a method calling `self._run_step` and `organize_files` to avoid the duplication.